src/app/redux/operations/ethrdid.ts

```diff
@@ -0,0 +1,82 @@
+import { Dispatch } from 'react'
+import EthrDID from '@rsksmart/ethr-did'
+import { getAccountAndNetwork } from '../../../ethrpc'
+import { changeOwner, resolveDid, addDelegate as addDelegateS } from '../reducers/ethrdid'
+import { getResolver } from 'ethr-did-resolver'
+import { DIDDocument, Resolver } from 'did-resolver'
+import { getSetting, SETTINGS } from '../../../config/getConfig'
+import { createDidFormat } from '../../../helpers'
+
+const Secp256k1VerificationKey2018 = '0x73696741757468'
+
+/**
+ * Returns the owner of a DID from the Ethr DID Registry
+ * @param provider web3 provider
+ */
+export const lookupOwner = (provider: any) => (dispatch: Dispatch<any>) =>
+  getAccountAndNetwork(provider).then(([address, chainId]) => {
+    new EthrDID({
+      address: address,
+      provider,
+      registry: getSetting(parseInt(chainId), SETTINGS.ETHR_DID_CONTRACT)
+    })
+      .lookupOwner(address).then((owner: string) => dispatch(changeOwner({ owner })))
+  })
+
+/**
+ * Set the owner in the DID Registry. Can only be set by the owner.
+ * @param provider web3 provider
+ * @param newOwner new owner of the DID
+ */
+export const setDidOwner = (provider: any, newOwner: string) => (dispatch: Dispatch<any>) =>
+  new Promise((resolve, reject) => {
+    getAccountAndNetwork(provider).then(([address, chainId]) =>
+      new EthrDID({
+        address: address,
+        provider,
+        registry: getSetting(parseInt(chainId), SETTINGS.ETHR_DID_CONTRACT)
+      })
+        .changeOwner(newOwner.toLowerCase())
+        .then(() => resolve(dispatch(changeOwner({ owner: newOwner }))))
+        .catch((err: Error) => reject(err))
+    )
+  })
+
+/**
+ * Reolve a DID
+ * @param provider web3 provider
+ */
+export const resolve = (provider: any) => (dispatch: Dispatch<any>) => {
+  getAccountAndNetwork(provider).then(([address, chainId]) => {
+    const providerConfig = {
+      networks: [
```

Comment:
I think the error I mentioned above is because we did not add Ethereum chain. We can use an Infura instance now

Reply:
We are using infura in the rLogin file and in the supported chains. I'm not sure why this error is here.

Reply:
We need to decide between:
- Adding the configuration for ethereum to the DID Resolver - _We are using infura in the rLogin file and in the supported chains. I'm not sure why this error is here,_. If tou are talking about https://github.com/rsksmart/rif-identity-manager/blob/main/src/rLogin.ts#L11, that is configuration only for Wallet Connect.
- Disable Ethereum from `supportedNetworks` rLogin feature -  _I'm adding in catches if not RSK or local to now show components_ that would not be necessary if the user never reaches the site with another network set

Reply:
I would go for enabling Ethereum with correct config :)